packages/neon-core/src/rpc/RPCClient.ts

```diff
@@ -220,10 +221,14 @@ export class RPCClient {
   /**
    * Gets the state of the contract at the given scriptHash.
    */
-  // TODO: parse result to manifest when contract manifest is merged
-  public async getContractState(scriptHash: string): Promise<object> {
+  public async getContractState(
+    scriptHash: string
+  ): Promise<ContractManifest | null> {
     const response = await this.execute(Query.getContractState(scriptHash));
-    return response.result;
+    if (response.error) {

```

Comment:
the rpc client will throw if there is an error. I dont think hiding the error is a good idea anyway.

Reply:
If we just catch the throwing error, we cannot tell exactly whether the error is because of network error or contract not exist.

If contract doesn't exist, it will return 200, too, with an error info in the response body. We can firmly tell user it doesn't exist.

Reply:
Ah i see my bad I misread the code. Maybe time to rework the RPC client a little.